design/parallel-commit/initial-design.md

```diff
@@ -17,7 +17,34 @@ This document is focussed on the initial increments of work to implement a corre
 
 Please see [this doc](./parallel-commit-known-issues-and-solutions.md).
 
-## Protocol design
+## Implementation
+
+### TiDB
+
+* The user should opt-in to parallel commit; we assume the user has opted in for all nodes for the rest of this document.
+* Each prewrite response will include a `min_commit_ts`, TiDB will select the largest `min_commit_ts` as the final `commit_ts` for the transaction.
+* TiDB can return success to the client before sending the commit message but after receiving success responses for all prewrite messages.
+* If an operation fails because it is locked, TiDB must query the primary lock to find a list of secondaries (a `CheckTxnStatus` request). It can then send a batch get request to each region to get all secondary locks and the `min_commit_ts` in the locks. If all locks are got, it can send the commit message as above. If any lock disappears, the primary key must have been committed or rolled back. Use `CheckTxnStatus` to get the updated information and resolve the locks.
```

Comment:
@sticnarf I'm not clear why TiDB uses a batch get request to see if the locks exist. I think that this would need some changes to the command. I think we might need a new command for this (or perhaps a new 'mode' for `ResolveLock`).

Reply:
`BatchGet` can be used to achive this goal. It looks tricky and a bit strange, however :)
Anyway, I'm also fine with another command for it.

Reply:
How does it work? I can imagine you can use RawBatchGet, but I don't see how BatchGet would work.

Reply:
`BatchGet` all secondary keys with max u64. It will return the locks if exists. https://github.com/tikv/tikv/blob/master/src/storage/mvcc/reader/point_getter.rs#L196